entity/contents/filters/filter_input.h

```diff
@@ -27,36 +32,119 @@ class Entity;
 ///
 /// Filters may decide to not evaluate inputs in situations where they won't
 /// contribute to the filter's output texture.
-class FilterInput final {
+class FilterInput {
  public:
   using Ref = std::shared_ptr<FilterInput>;
   using Vector = std::vector<FilterInput::Ref>;
-  using Variant =
-      std::variant<std::shared_ptr<Texture>, std::shared_ptr<Contents>>;
+  using Variant = std::variant<std::shared_ptr<FilterContents>,
+                               std::shared_ptr<Contents>,
+                               std::shared_ptr<Texture>>;
 
-  ~FilterInput();
+  virtual ~FilterInput();
 
   static FilterInput::Ref Make(Variant input);
 
   static FilterInput::Vector Make(std::initializer_list<Variant> inputs);
 
-  Variant GetInput() const;
+  virtual Variant GetInput() const = 0;
+
+  virtual std::optional<Snapshot> GetSnapshot(const ContentContext& renderer,
+                                              const Entity& entity) const = 0;
+
+  virtual std::optional<Rect> GetCoverage(const Entity& entity) const = 0;
+
+  virtual Matrix GetLocalTransform(const Entity& entity) const;
```

Comment:
A brief explanation in a headerdoc of the differences between a local transform and the transform would be good.

Reply:
Done.